HTTP server cannot be started due to Context intialization failure

Citrus Version: at least 3.2.0 & 3.2.1

---

Expected behavior: 

When a fat jar is built, started and then triggered, the http server should be started.

---

Actual behavior:

Tests are skipped, and an error is shown:
```
2022-07-07 15:27:40.436 ERROR --- [pool-2-thread-1] c.c.c.h.s.CitrusDispatcherServlet        : Context initialization failed
2022-07-07T15:27:40.439671823Z org.springframework.beans.factory.parsing.BeanDefinitionParsingException: Configuration problem: Unable to locate Spring NamespaceHandler for XML schema namespace [http://www.springframework.org/schema/context]
2022-07-07T15:27:40.439680911Z Offending resource: class path resource [com/consol/citrus/http/citrus-servlet-context.xml]
2022-07-07T15:27:40.439688175Z 
2022-07-07T15:27:40.439694296Z 	at org.springframework.beans.factory.parsing.FailFastProblemReporter.error(FailFastProblemReporter.java:72)
2022-07-07T15:27:40.439700278Z 	at org.springframework.beans.factory.parsing.ReaderContext.error(ReaderContext.java:119)
2022-07-07T15:27:40.439707492Z 	at org.springframework.beans.factory.parsing.ReaderContext.error(ReaderContext.java:111)
2022-07-07T15:27:40.439715537Z 	at org.springframework.beans.factory.xml.BeanDefinitionParserDelegate.error(BeanDefinitionParserDelegate.java:281)
2022-07-07T15:27:40.439737780Z 	at org.springframework.beans.factory.xml.BeanDefinitionParserDelegate.parseCustomElement(BeanDefinitionParserDelegate.java:1388)
2022-07-07T15:27:40.439744352Z 	at org.springframework.beans.factory.xml.BeanDefinitionParserDelegate.parseCustomElement(BeanDefinitionParserDelegate.java:1371)
2022-07-07T15:27:40.439750284Z 	at org.springframework.beans.factory.xml.DefaultBeanDefinitionDocumentReader.parseBeanDefinitions(DefaultBeanDefinitionDocumentReader.java:179)
2022-07-07T15:27:40.439756486Z 	at org.springframework.beans.factory.xml.DefaultBeanDefinitionDocumentReader.doRegisterBeanDefinitions(DefaultBeanDefinitionDocumentReader.java:149)
2022-07-07T15:27:40.439762798Z 	at org.springframework.beans.factory.xml.DefaultBeanDefinitionDocumentReader.registerBeanDefinitions(DefaultBeanDefinitionDocumentReader.java:96)
2022-07-07T15:27:40.439770693Z 	at org.springframework.beans.factory.xml.XmlBeanDefinitionReader.registerBeanDefinitions(XmlBeanDefinitionReader.java:511)
2022-07-07T15:27:40.439782776Z 	at org.springframework.beans.factory.xml.XmlBeanDefinitionReader.doLoadBeanDefinitions(XmlBeanDefinitionReader.java:391)
2022-07-07T15:27:40.439790892Z 	at org.springframework.beans.factory.xml.XmlBeanDefinitionReader.loadBeanDefinitions(XmlBeanDefinitionReader.java:338)
2022-07-07T15:27:40.439797685Z 	at org.springframework.beans.factory.xml.XmlBeanDefinitionReader.loadBeanDefinitions(XmlBeanDefinitionReader.java:310)
2022-07-07T15:27:40.439806492Z 	at org.springframework.beans.factory.support.AbstractBeanDefinitionReader.loadBeanDefinitions(AbstractBeanDefinitionReader.java:196)
2022-07-07T15:27:40.439812403Z 	at org.springframework.beans.factory.support.AbstractBeanDefinitionReader.loadBeanDefinitions(AbstractBeanDefinitionReader.java:232)
2022-07-07T15:27:40.439818375Z 	at org.springframework.beans.factory.support.AbstractBeanDefinitionReader.loadBeanDefinitions(AbstractBeanDefinitionReader.java:203)
2022-07-07T15:27:40.439824767Z 	at org.springframework.web.context.support.XmlWebApplicationContext.loadBeanDefinitions(XmlWebApplicationContext.java:125)
2022-07-07T15:27:40.449548764Z 	at org.springframework.web.context.support.XmlWebApplicationContext.loadBeanDefinitions(XmlWebApplicationContext.java:94)
2022-07-07T15:27:40.449664286Z 	at org.springframework.context.support.AbstractRefreshableApplicationContext.refreshBeanFactory(AbstractRefreshableApplicationContext.java:130)
2022-07-07T15:27:40.449763165Z 	at org.springframework.context.support.AbstractApplicationContext.obtainFreshBeanFactory(AbstractApplicationContext.java:671)
2022-07-07T15:27:40.449857216Z 	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:553)
2022-07-07T15:27:40.449969892Z 	at org.springframework.web.servlet.FrameworkServlet.configureAndRefreshWebApplicationContext(FrameworkServlet.java:702)
2022-07-07T15:27:40.450081216Z 	at org.springframework.web.servlet.FrameworkServlet.createWebApplicationContext(FrameworkServlet.java:668)
2022-07-07T15:27:40.450191408Z 	at org.springframework.web.servlet.FrameworkServlet.createWebApplicationContext(FrameworkServlet.java:716)
2022-07-07T15:27:40.450333661Z 	at org.springframework.web.servlet.FrameworkServlet.initWebApplicationContext(FrameworkServlet.java:591)
2022-07-07T15:27:40.451071457Z 	at org.springframework.web.servlet.FrameworkServlet.initServletBean(FrameworkServlet.java:530)
2022-07-07T15:27:40.451332479Z 	at org.springframework.web.servlet.HttpServletBean.init(HttpServletBean.java:170)
2022-07-07T15:27:40.451453591Z 	at javax.servlet.GenericServlet.init(GenericServlet.java:180)
2022-07-07T15:27:40.451555056Z 	at org.eclipse.jetty.servlet.ServletHolder.initServlet(ServletHolder.java:632)
2022-07-07T15:27:40.451652453Z 	at org.eclipse.jetty.servlet.ServletHolder.initialize(ServletHolder.java:415)
2022-07-07T15:27:40.451752255Z 	at org.eclipse.jetty.servlet.ServletHandler.lambda$initialize$0(ServletHandler.java:731)
2022-07-07T15:27:40.451877706Z 	at java.base/java.util.stream.SortedOps$SizedRefSortingSink.end(SortedOps.java:357)
2022-07-07T15:27:40.451976335Z 	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:510)
2022-07-07T15:27:40.452075515Z 	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
2022-07-07T15:27:40.452173263Z 	at java.base/java.util.stream.StreamSpliterators$WrappingSpliterator.forEachRemaining(StreamSpliterators.java:310)
2022-07-07T15:27:40.452276030Z 	at java.base/java.util.stream.Streams$ConcatSpliterator.forEachRemaining(Streams.java:735)
2022-07-07T15:27:40.452380010Z 	at java.base/java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:762)
2022-07-07T15:27:40.452480423Z 	at org.eclipse.jetty.servlet.ServletHandler.initialize(ServletHandler.java:755)
2022-07-07T15:27:40.452576638Z 	at org.eclipse.jetty.servlet.ServletContextHandler.startContext(ServletContextHandler.java:379)
2022-07-07T15:27:40.452674325Z 	at org.eclipse.jetty.server.handler.ContextHandler.doStart(ContextHandler.java:910)
2022-07-07T15:27:40.452771181Z 	at org.eclipse.jetty.servlet.ServletContextHandler.doStart(ServletContextHandler.java:288)
2022-07-07T15:27:40.452895460Z 	at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:73)
2022-07-07T15:27:40.452990843Z 	at org.eclipse.jetty.util.component.ContainerLifeCycle.start(ContainerLifeCycle.java:169)
2022-07-07T15:27:40.453087679Z 	at org.eclipse.jetty.util.component.ContainerLifeCycle.doStart(ContainerLifeCycle.java:117)
2022-07-07T15:27:40.453190536Z 	at org.eclipse.jetty.server.handler.AbstractHandler.doStart(AbstractHandler.java:97)
2022-07-07T15:27:40.453323261Z 	at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:73)
2022-07-07T15:27:40.453424365Z 	at org.eclipse.jetty.util.component.ContainerLifeCycle.start(ContainerLifeCycle.java:169)
2022-07-07T15:27:40.453520470Z 	at org.eclipse.jetty.util.component.ContainerLifeCycle.doStart(ContainerLifeCycle.java:117)
2022-07-07T15:27:40.453617596Z 	at org.eclipse.jetty.server.handler.AbstractHandler.doStart(AbstractHandler.java:97)
2022-07-07T15:27:40.453712398Z 	at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:73)
2022-07-07T15:27:40.453823191Z 	at org.eclipse.jetty.util.component.ContainerLifeCycle.start(ContainerLifeCycle.java:169)
2022-07-07T15:27:40.453943302Z 	at org.eclipse.jetty.server.Server.start(Server.java:423)
2022-07-07T15:27:40.454040959Z 	at org.eclipse.jetty.util.component.ContainerLifeCycle.doStart(ContainerLifeCycle.java:110)
2022-07-07T15:27:40.454137264Z 	at org.eclipse.jetty.server.handler.AbstractHandler.doStart(AbstractHandler.java:97)
2022-07-07T15:27:40.454237486Z 	at org.eclipse.jetty.server.Server.doStart(Server.java:387)
2022-07-07T15:27:40.454343139Z 	at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:73)
2022-07-07T15:27:40.454444524Z 	at com.consol.citrus.http.server.HttpServer.startup(HttpServer.java:210)
2022-07-07T15:27:40.454543504Z 	at com.consol.citrus.server.AbstractServer.start(AbstractServer.java:93)
2022-07-07T15:27:40.454638977Z 	at com.consol.citrus.server.AbstractServer.initialize(AbstractServer.java:165)
2022-07-07T15:27:40.454734871Z 	at com.consol.citrus.http.server.HttpServer.initialize(HttpServer.java:219)
2022-07-07T15:27:40.454832379Z 	at com.consol.citrus.config.ComponentLifecycleProcessor.postProcessBeforeInitialization(ComponentLifecycleProcessor.java:59)
2022-07-07T15:27:40.454956667Z 	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.applyBeanPostProcessorsBeforeInitialization(AbstractAutowireCapableBeanFactory.java:440)
2022-07-07T15:27:40.455053373Z 	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1796)
2022-07-07T15:27:40.455148987Z 	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:620)
2022-07-07T15:27:40.455246564Z 	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:542)
2022-07-07T15:27:40.455355513Z 	at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:335)
2022-07-07T15:27:40.455472578Z 	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:234)
2022-07-07T15:27:40.455598119Z 	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:333)
2022-07-07T15:27:40.455707399Z 	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:208)
2022-07-07T15:27:40.455809144Z 	at org.springframework.beans.factory.support.DefaultListableBeanFactory.preInstantiateSingletons(DefaultListableBeanFactory.java:953)
2022-07-07T15:27:40.455949103Z 	at org.springframework.context.support.AbstractApplicationContext.finishBeanFactoryInitialization(AbstractApplicationContext.java:918)
2022-07-07T15:27:40.456051309Z 	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:583)
2022-07-07T15:27:40.456171730Z 	at org.springframework.context.annotation.AnnotationConfigApplicationContext.<init>(AnnotationConfigApplicationContext.java:93)
2022-07-07T15:27:40.456274518Z 	at com.consol.citrus.CitrusSpringContext.create(CitrusSpringContext.java:67)
2022-07-07T15:27:40.456375241Z 	at com.consol.citrus.CitrusSpringContextProvider.create(CitrusSpringContextProvider.java:38)
2022-07-07T15:27:40.456484100Z 	at com.consol.citrus.CitrusInstanceManager.newInstance(CitrusInstanceManager.java:45)
2022-07-07T15:27:40.456587749Z 	at com.consol.citrus.Citrus.newInstance(Citrus.java:65)
2022-07-07T15:27:40.456689124Z 	at com.consol.citrus.testng.spring.TestNGCitrusSpringSupport.before(TestNGCitrusSpringSupport.java:176)
2022-07-07T15:27:40.456743695Z 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
2022-07-07T15:27:40.456760127Z 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
2022-07-07T15:27:40.456766129Z 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
2022-07-07T15:27:40.456770146Z 	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
2022-07-07T15:27:40.456774134Z 	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:132)
2022-07-07T15:27:40.456778262Z 	at org.testng.internal.MethodInvocationHelper.invokeMethodConsideringTimeout(MethodInvocationHelper.java:61)
2022-07-07T15:27:40.456809973Z 	at org.testng.internal.ConfigInvoker.invokeConfigurationMethod(ConfigInvoker.java:366)
2022-07-07T15:27:40.456817016Z 	at org.testng.internal.ConfigInvoker.invokeConfigurations(ConfigInvoker.java:320)
2022-07-07T15:27:40.456821385Z 	at org.testng.internal.TestMethodWorker.invokeBeforeClassMethods(TestMethodWorker.java:176)
2022-07-07T15:27:40.456825072Z 	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:122)
2022-07-07T15:27:40.456828639Z 	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
2022-07-07T15:27:40.456832306Z 	at org.testng.TestRunner.privateRun(TestRunner.java:764)
2022-07-07T15:27:40.456835943Z 	at org.testng.TestRunner.run(TestRunner.java:585)
2022-07-07T15:27:40.456839730Z 	at org.testng.SuiteRunner.runTest(SuiteRunner.java:384)
2022-07-07T15:27:40.456843397Z 	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:378)
2022-07-07T15:27:40.456872994Z 	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:337)
2022-07-07T15:27:40.456877813Z 	at org.testng.SuiteRunner.run(SuiteRunner.java:286)
2022-07-07T15:27:40.456881390Z 	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:53)
2022-07-07T15:27:40.456886259Z 	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:96)
2022-07-07T15:27:40.456889826Z 	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1218)
2022-07-07T15:27:40.456893373Z 	at org.testng.TestNG.runSuitesLocally(TestNG.java:1140)
2022-07-07T15:27:40.456913071Z 	at org.testng.TestNG.runSuites(TestNG.java:1069)
2022-07-07T15:27:40.456920255Z 	at org.testng.TestNG.run(TestNG.java:1037)
2022-07-07T15:27:40.456924553Z 	at com.consol.citrus.testng.TestNGEngine.run(TestNGEngine.java:151)
2022-07-07T15:27:40.456928320Z 	at com.consol.citrus.main.CitrusApp.run(CitrusApp.java:125)
2022-07-07T15:27:40.456931887Z 	at com.consol.citrus.remote.controller.RunController.run(RunController.java:107)
2022-07-07T15:27:40.456935303Z 	at com.consol.citrus.remote.controller.RunController.runPackages(RunController.java:75)
2022-07-07T15:27:40.456938930Z 	at com.consol.citrus.remote.controller.RunController.runAll(RunController.java:60)
2022-07-07T15:27:40.456942587Z 	at com.consol.citrus.remote.CitrusRemoteApplication.runTests(CitrusRemoteApplication.java:262)
2022-07-07T15:27:40.456946535Z 	at com.consol.citrus.remote.CitrusRemoteApplication$1.run(CitrusRemoteApplication.java:215)
2022-07-07T15:27:40.456950182Z 	at com.consol.citrus.remote.job.RunJob.call(RunJob.java:43)
2022-07-07T15:27:40.456953729Z 	at com.consol.citrus.remote.job.RunJob.call(RunJob.java:29)
2022-07-07T15:27:40.456957416Z 	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
2022-07-07T15:27:40.456960973Z 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
2022-07-07T15:27:40.456964650Z 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
2022-07-07T15:27:40.456968347Z 	at java.base/java.lang.Thread.run(Thread.java:833)
2022-07-07T15:27:40.456971984Z 2022-07-07 15:27:40.440  WARN --- [pool-2-thread-1] o.e.j.s.h.C.ROOT                         : unavailable
2022-07-07T15:27:40.456976232Z org.springframework.beans.factory.parsing.BeanDefinitionParsingException: Configuration problem: Unable to locate Spring NamespaceHandler for XML schema namespace [http://www.springframework.org/schema/context]
2022-07-07T15:27:40.456980119Z Offending resource: class path resource [com/consol/citrus/http/citrus-servlet-context.xml]
2022-07-07T15:27:40.456983807Z 
2022-07-07T15:27:40.456987123Z 	at org.springframework.beans.factory.parsing.FailFastProblemReporter.error(FailFastProblemReporter.java:72)
2022-07-07T15:27:40.456990900Z 	at org.springframework.beans.factory.parsing.ReaderContext.error(ReaderContext.java:119)
2022-07-07T15:27:40.456994527Z 	at org.springframework.beans.factory.parsing.ReaderContext.error(ReaderContext.java:111)
2022-07-07T15:27:40.456997924Z 	at org.springframework.beans.factory.xml.BeanDefinitionParserDelegate.error(BeanDefinitionParserDelegate.java:281)
2022-07-07T15:27:40.457001911Z 	at org.springframework.beans.factory.xml.BeanDefinitionParserDelegate.parseCustomElement(BeanDefinitionParserDelegate.java:1388)
2022-07-07T15:27:40.457006450Z 	at org.springframework.beans.factory.xml.BeanDefinitionParserDelegate.parseCustomElement(BeanDefinitionParserDelegate.java:1371)
2022-07-07T15:27:40.457010498Z 	at org.springframework.beans.factory.xml.DefaultBeanDefinitionDocumentReader.parseBeanDefinitions(DefaultBeanDefinitionDocumentReader.java:179)
2022-07-07T15:27:40.457019495Z 	at org.springframework.beans.factory.xml.DefaultBeanDefinitionDocumentReader.doRegisterBeanDefinitions(DefaultBeanDefinitionDocumentReader.java:149)
2022-07-07T15:27:40.457023553Z 	at org.springframework.beans.factory.xml.DefaultBeanDefinitionDocumentReader.registerBeanDefinitions(DefaultBeanDefinitionDocumentReader.java:96)
2022-07-07T15:27:40.457027250Z 	at org.springframework.beans.factory.xml.XmlBeanDefinitionReader.registerBeanDefinitions(XmlBeanDefinitionReader.java:511)
2022-07-07T15:27:40.457030767Z 	at org.springframework.beans.factory.xml.XmlBeanDefinitionReader.doLoadBeanDefinitions(XmlBeanDefinitionReader.java:391)
2022-07-07T15:27:40.457060804Z 	at org.springframework.beans.factory.xml.XmlBeanDefinitionReader.loadBeanDefinitions(XmlBeanDefinitionReader.java:338)
2022-07-07T15:27:40.457069070Z 	at org.springframework.beans.factory.xml.XmlBeanDefinitionReader.loadBeanDefinitions(XmlBeanDefinitionReader.java:310)
2022-07-07T15:27:40.457073298Z 	at org.springframework.beans.factory.support.AbstractBeanDefinitionReader.loadBeanDefinitions(AbstractBeanDefinitionReader.java:196)
2022-07-07T15:27:40.457077246Z 	at org.springframework.beans.factory.support.AbstractBeanDefinitionReader.loadBeanDefinitions(AbstractBeanDefinitionReader.java:232)
2022-07-07T15:27:40.457081214Z 	at org.springframework.beans.factory.support.AbstractBeanDefinitionReader.loadBeanDefinitions(AbstractBeanDefinitionReader.java:203)
2022-07-07T15:27:40.457084881Z 	at org.springframework.web.context.support.XmlWebApplicationContext.loadBeanDefinitions(XmlWebApplicationContext.java:125)
2022-07-07T15:27:40.457088638Z 	at org.springframework.web.context.support.XmlWebApplicationContext.loadBeanDefinitions(XmlWebApplicationContext.java:94)
2022-07-07T15:27:40.457092395Z 	at org.springframework.context.support.AbstractRefreshableApplicationContext.refreshBeanFactory(AbstractRefreshableApplicationContext.java:130)
2022-07-07T15:27:40.457096363Z 	at org.springframework.context.support.AbstractApplicationContext.obtainFreshBeanFactory(AbstractApplicationContext.java:671)
2022-07-07T15:27:40.457100160Z 	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:553)
2022-07-07T15:27:40.457106001Z 	at org.springframework.web.servlet.FrameworkServlet.configureAndRefreshWebApplicationContext(FrameworkServlet.java:702)
2022-07-07T15:27:40.457109979Z 	at org.springframework.web.servlet.FrameworkServlet.createWebApplicationContext(FrameworkServlet.java:668)
2022-07-07T15:27:40.457114928Z 	at org.springframework.web.servlet.FrameworkServlet.createWebApplicationContext(FrameworkServlet.java:716)
2022-07-07T15:27:40.457118525Z 	at org.springframework.web.servlet.FrameworkServlet.initWebApplicationContext(FrameworkServlet.java:591)
2022-07-07T15:27:40.457121862Z 	at org.springframework.web.servlet.FrameworkServlet.initServletBean(FrameworkServlet.java:530)
2022-07-07T15:27:40.457125619Z 	at org.springframework.web.servlet.HttpServletBean.init(HttpServletBean.java:170)
2022-07-07T15:27:40.457134887Z 	at javax.servlet.GenericServlet.init(GenericServlet.java:180)
2022-07-07T15:27:40.457139065Z 	at org.eclipse.jetty.servlet.ServletHolder.initServlet(ServletHolder.java:632)
2022-07-07T15:27:40.457142832Z 	at org.eclipse.jetty.servlet.ServletHolder.initialize(ServletHolder.java:415)
2022-07-07T15:27:40.457146449Z 	at org.eclipse.jetty.servlet.ServletHandler.lambda$initialize$0(ServletHandler.java:731)
2022-07-07T15:27:40.457150006Z 	at java.base/java.util.stream.SortedOps$SizedRefSortingSink.end(SortedOps.java:357)
2022-07-07T15:27:40.457153532Z 	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:510)
2022-07-07T15:27:40.457157199Z 	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
2022-07-07T15:27:40.457160877Z 	at java.base/java.util.stream.StreamSpliterators$WrappingSpliterator.forEachRemaining(StreamSpliterators.java:310)
2022-07-07T15:27:40.457164774Z 	at java.base/java.util.stream.Streams$ConcatSpliterator.forEachRemaining(Streams.java:735)
2022-07-07T15:27:40.457168431Z 	at java.base/java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:762)
2022-07-07T15:27:40.457172138Z 	at org.eclipse.jetty.servlet.ServletHandler.initialize(ServletHandler.java:755)
2022-07-07T15:27:40.457175705Z 	at org.eclipse.jetty.servlet.ServletContextHandler.startContext(ServletContextHandler.java:379)
2022-07-07T15:27:40.457179502Z 	at org.eclipse.jetty.server.handler.ContextHandler.doStart(ContextHandler.java:910)
2022-07-07T15:27:40.457183169Z 	at org.eclipse.jetty.servlet.ServletContextHandler.doStart(ServletContextHandler.java:288)
2022-07-07T15:27:40.457186786Z 	at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:73)
2022-07-07T15:27:40.457190273Z 	at org.eclipse.jetty.util.component.ContainerLifeCycle.start(ContainerLifeCycle.java:169)
2022-07-07T15:27:40.457194772Z 	at org.eclipse.jetty.util.component.ContainerLifeCycle.doStart(ContainerLifeCycle.java:117)
2022-07-07T15:27:40.457198549Z 	at org.eclipse.jetty.server.handler.AbstractHandler.doStart(AbstractHandler.java:97)
2022-07-07T15:27:40.457202336Z 	at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:73)
2022-07-07T15:27:40.457206224Z 	at org.eclipse.jetty.util.component.ContainerLifeCycle.start(ContainerLifeCycle.java:169)
2022-07-07T15:27:40.457209911Z 	at org.eclipse.jetty.util.component.ContainerLifeCycle.doStart(ContainerLifeCycle.java:117)
2022-07-07T15:27:40.457213528Z 	at org.eclipse.jetty.server.handler.AbstractHandler.doStart(AbstractHandler.java:97)
2022-07-07T15:27:40.457217074Z 	at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:73)
2022-07-07T15:27:40.457220571Z 	at org.eclipse.jetty.util.component.ContainerLifeCycle.start(ContainerLifeCycle.java:169)
2022-07-07T15:27:40.457224068Z 	at org.eclipse.jetty.server.Server.start(Server.java:423)
2022-07-07T15:27:40.457231612Z 	at org.eclipse.jetty.util.component.ContainerLifeCycle.doStart(ContainerLifeCycle.java:110)
2022-07-07T15:27:40.457235500Z 	at org.eclipse.jetty.server.handler.AbstractHandler.doStart(AbstractHandler.java:97)
2022-07-07T15:27:40.457239217Z 	at org.eclipse.jetty.server.Server.doStart(Server.java:387)
2022-07-07T15:27:40.457242674Z 	at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:73)
2022-07-07T15:27:40.457246250Z 	at com.consol.citrus.http.server.HttpServer.startup(HttpServer.java:210)
2022-07-07T15:27:40.457249807Z 	at com.consol.citrus.server.AbstractServer.start(AbstractServer.java:93)
2022-07-07T15:27:40.457253364Z 	at com.consol.citrus.server.AbstractServer.initialize(AbstractServer.java:165)
2022-07-07T15:27:40.457256871Z 	at com.consol.citrus.http.server.HttpServer.initialize(HttpServer.java:219)
2022-07-07T15:27:40.457260358Z 	at com.consol.citrus.config.ComponentLifecycleProcessor.postProcessBeforeInitialization(ComponentLifecycleProcessor.java:59)
2022-07-07T15:27:40.457264255Z 	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.applyBeanPostProcessorsBeforeInitialization(AbstractAutowireCapableBeanFactory.java:440)
2022-07-07T15:27:40.457268183Z 	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1796)
2022-07-07T15:27:40.457271749Z 	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:620)
2022-07-07T15:27:40.457276218Z 	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:542)
2022-07-07T15:27:40.457280035Z 	at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:335)
2022-07-07T15:27:40.457283712Z 	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:234)
2022-07-07T15:27:40.457287430Z 	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:333)
2022-07-07T15:27:40.457290846Z 	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:208)
2022-07-07T15:27:40.457294623Z 	at org.springframework.beans.factory.support.DefaultListableBeanFactory.preInstantiateSingletons(DefaultListableBeanFactory.java:953)
2022-07-07T15:27:40.457298330Z 	at org.springframework.context.support.AbstractApplicationContext.finishBeanFactoryInitialization(AbstractApplicationContext.java:918)
2022-07-07T15:27:40.457302098Z 	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:583)
2022-07-07T15:27:40.457305745Z 	at org.springframework.context.annotation.AnnotationConfigApplicationContext.<init>(AnnotationConfigApplicationContext.java:93)
2022-07-07T15:27:40.457309752Z 	at com.consol.citrus.CitrusSpringContext.create(CitrusSpringContext.java:67)
2022-07-07T15:27:40.457313279Z 	at com.consol.citrus.CitrusSpringContextProvider.create(CitrusSpringContextProvider.java:38)
2022-07-07T15:27:40.457321074Z 	at com.consol.citrus.CitrusInstanceManager.newInstance(CitrusInstanceManager.java:45)
2022-07-07T15:27:40.457325002Z 	at com.consol.citrus.Citrus.newInstance(Citrus.java:65)
2022-07-07T15:27:40.457328679Z 	at com.consol.citrus.testng.spring.TestNGCitrusSpringSupport.before(TestNGCitrusSpringSupport.java:176)
2022-07-07T15:27:40.457332276Z 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
2022-07-07T15:27:40.457335822Z 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
2022-07-07T15:27:40.457339399Z 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
2022-07-07T15:27:40.457343126Z 	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
2022-07-07T15:27:40.457346443Z 	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:132)
2022-07-07T15:27:40.457350070Z 	at org.testng.internal.MethodInvocationHelper.invokeMethodConsideringTimeout(MethodInvocationHelper.java:61)
2022-07-07T15:27:40.457353787Z 	at org.testng.internal.ConfigInvoker.invokeConfigurationMethod(ConfigInvoker.java:366)
2022-07-07T15:27:40.457358686Z 	at org.testng.internal.ConfigInvoker.invokeConfigurations(ConfigInvoker.java:320)
2022-07-07T15:27:40.457362383Z 	at org.testng.internal.TestMethodWorker.invokeBeforeClassMethods(TestMethodWorker.java:176)
2022-07-07T15:27:40.457365910Z 	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:122)
2022-07-07T15:27:40.457369357Z 	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
2022-07-07T15:27:40.457372854Z 	at org.testng.TestRunner.privateRun(TestRunner.java:764)
2022-07-07T15:27:40.457376480Z 	at org.testng.TestRunner.run(TestRunner.java:585)
2022-07-07T15:27:40.457380047Z 	at org.testng.SuiteRunner.runTest(SuiteRunner.java:384)
2022-07-07T15:27:40.457383634Z 	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:378)
2022-07-07T15:27:40.457387141Z 	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:337)
2022-07-07T15:27:40.457390628Z 	at org.testng.SuiteRunner.run(SuiteRunner.java:286)
2022-07-07T15:27:40.457394134Z 	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:53)
2022-07-07T15:27:40.457397681Z 	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:96)
2022-07-07T15:27:40.457401388Z 	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1218)
2022-07-07T15:27:40.457404865Z 	at org.testng.TestNG.runSuitesLocally(TestNG.java:1140)
2022-07-07T15:27:40.457408372Z 	at org.testng.TestNG.runSuites(TestNG.java:1069)
2022-07-07T15:27:40.457411909Z 	at org.testng.TestNG.run(TestNG.java:1037)
2022-07-07T15:27:40.457415475Z 	at com.consol.citrus.testng.TestNGEngine.run(TestNGEngine.java:151)
2022-07-07T15:27:40.457423511Z 	at com.consol.citrus.main.CitrusApp.run(CitrusApp.java:125)
2022-07-07T15:27:40.457427408Z 	at com.consol.citrus.remote.controller.RunController.run(RunController.java:107)
2022-07-07T15:27:40.457430955Z 	at com.consol.citrus.remote.controller.RunController.runPackages(RunController.java:75)
2022-07-07T15:27:40.457434562Z 	at com.consol.citrus.remote.controller.RunController.runAll(RunController.java:60)
2022-07-07T15:27:40.457438219Z 	at com.consol.citrus.remote.CitrusRemoteApplication.runTests(CitrusRemoteApplication.java:262)
2022-07-07T15:27:40.457442267Z 	at com.consol.citrus.remote.CitrusRemoteApplication$1.run(CitrusRemoteApplication.java:215)
2022-07-07T15:27:40.457445713Z 	at com.consol.citrus.remote.job.RunJob.call(RunJob.java:43)
2022-07-07T15:27:40.457449350Z 	at com.consol.citrus.remote.job.RunJob.call(RunJob.java:29)
2022-07-07T15:27:40.457453017Z 	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
2022-07-07T15:27:40.457456534Z 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
2022-07-07T15:27:40.457460131Z 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
2022-07-07T15:27:40.457463748Z 	at java.base/java.lang.Thread.run(Thread.java:833)
2022-07-07T15:27:40.460245673Z 2022-07-07 15:27:40.457  WARN --- [pool-2-thread-1] s.c.a.AnnotationConfigApplicationContext : Exception encountered during context initialization - cancelling refresh attempt: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'fooServer' defined in class path resource [de/turing85/citrus/tests/citrus/configuration/HttpServers.class]: Initialization of bean failed; nested exception is com.consol.citrus.exceptions.CitrusRuntimeException: javax.servlet.ServletException: fooServer-servlet==com.consol.citrus.http.servlet.CitrusDispatcherServlet@6b7c7821{jsp=null,order=-1,inst=true,async=true,src=EMBEDDED:null,STARTED}
2022-07-07T15:27:40.563627961Z 
2022-07-07T15:27:40.563678337Z ===============================================
2022-07-07T15:27:40.563692665Z Default Suite
2022-07-07T15:27:40.563698717Z Total tests run: 1, Passes: 0, Failures: 0, Skips: 1
2022-07-07T15:27:40.563702404Z Configuration Failures: 3, Skips: 0
2022-07-07T15:27:40.563706972Z ===============================================
2022-07-07T15:27:40.563710209Z 
2022-07-07T15:27:40.902364645Z 2022-07-07 15:27:40.901  INFO --- [tp1875706559-24] c.c.c.r.CitrusRemoteApplication          : GET http://localhost:4567/results/files
2022-07-07T15:27:40.912969404Z 2022-07-07 15:27:40.912  INFO --- [tp1875706559-25] c.c.c.r.CitrusRemoteApplication          : GET http://localhost:4567/results/suite
2022-07-07T15:27:40.924088634Z 2022-07-07 15:27:40.923  INFO --- [tp1875706559-24] c.c.c.r.CitrusRemoteApplication          : GET http://localhost:4567/results/file/TEST-de.turing85.citrus.tests.citrus.GetHelloIT.xml
```
---

Reproducer:

- checkout https://github.com/turing85/citrus-playground/tree/http-server-not-working
- build both applications (sut and citrus tests): `./mwnv  -P build-test-jar package`
- deploy the test setup via `docker compose`: `cd local-deployment && docker compose up -d --build && cd ..`
- start the tests through maven directly: `./mvnw -f citrus verify`
- observe that the tests succeed
- start the tests deployed in docker through the remote-plugin. `./mvnw -f citrus -P trigger-citrus-remote verify`
- observe that the tests fail. The above error can be found in the logs of the `citrus-tests` container.

Update: after adding 
```
<dependency>
    <groupId>org.springframework</groupId>
    <artifactId>spring-context</artifactId>
    <version>${spring.version}</version>
</dependency>
```

The error changes to 

```
2022-07-07 17:18:07.120 ERROR --- [pool-2-thread-1] c.c.c.h.s.CitrusDispatcherServlet        : Context initialization failed
2022-07-07T17:18:07.128032434Z org.springframework.beans.factory.parsing.BeanDefinitionParsingException: Configuration problem: Unexpected failure during bean definition parsing
2022-07-07T17:18:07.128041772Z Offending resource: class path resource [com/consol/citrus/http/citrus-servlet-context.xml]
2022-07-07T17:18:07.128046771Z Bean 'citrusMethodHandlerAdapter'; nested exception is org.springframework.beans.factory.parsing.BeanDefinitionParsingException: Configuration problem: Unable to locate Spring NamespaceHandler for XML schema namespace [http://www.springframework.org/schema/util]
2022-07-07T17:18:07.128098621Z Offending resource: class path resource [com/consol/citrus/http/citrus-servlet-context.xml]
2022-07-07T17:18:07.128106837Z Property 'messageConverters'
2022-07-07T17:18:07.128111025Z 	-> Bean 'citrusMethodHandlerAdapter'
2022-07-07T17:18:07.128115183Z 	at org.springframework.beans.factory.parsing.FailFastProblemReporter.error(FailFastProblemReporter.java:72)
2022-07-07T17:18:07.128119250Z 	at org.springframework.beans.factory.parsing.ReaderContext.error(ReaderContext.java:119)
2022-07-07T17:18:07.128123438Z 	at org.springframework.beans.factory.xml.BeanDefinitionParserDelegate.error(BeanDefinitionParserDelegate.java:288)
2022-07-07T17:18:07.128127516Z 	at org.springframework.beans.factory.xml.BeanDefinitionParserDelegate.parseBeanDefinitionElement(BeanDefinitionParserDelegate.java:540)
2022-07-07T17:18:07.128131314Z 	at org.springframework.beans.factory.xml.BeanDefinitionParserDelegate.parseBeanDefinitionElement(BeanDefinitionParserDelegate.java:437)
2022-07-07T17:18:07.128146803Z 	at org.springframework.beans.factory.xml.BeanDefinitionParserDelegate.parseBeanDefinitionElement(BeanDefinitionParserDelegate.java:405)
2022-07-07T17:18:07.128151272Z 	at org.springframework.beans.factory.xml.DefaultBeanDefinitionDocumentReader.processBeanDefinition(DefaultBeanDefinitionDocumentReader.java:306)
2022-07-07T17:18:07.128155370Z 	at org.springframework.beans.factory.xml.DefaultBeanDefinitionDocumentReader.parseDefaultElement(DefaultBeanDefinitionDocumentReader.java:197)
2022-07-07T17:18:07.128159728Z 	at org.springframework.beans.factory.xml.DefaultBeanDefinitionDocumentReader.parseBeanDefinitions(DefaultBeanDefinitionDocumentReader.java:176)
2022-07-07T17:18:07.128165058Z 	at org.springframework.beans.factory.xml.DefaultBeanDefinitionDocumentReader.doRegisterBeanDefinitions(DefaultBeanDefinitionDocumentReader.java:149)
2022-07-07T17:18:07.128169477Z 	at org.springframework.beans.factory.xml.DefaultBeanDefinitionDocumentReader.registerBeanDefinitions(DefaultBeanDefinitionDocumentReader.java:96)
2022-07-07T17:18:07.128173454Z 	at org.springframework.beans.factory.xml.XmlBeanDefinitionReader.registerBeanDefinitions(XmlBeanDefinitionReader.java:511)
2022-07-07T17:18:07.128177382Z 	at org.springframework.beans.factory.xml.XmlBeanDefinitionReader.doLoadBeanDefinitions(XmlBeanDefinitionReader.java:391)
2022-07-07T17:18:07.128181370Z 	at org.springframework.beans.factory.xml.XmlBeanDefinitionReader.loadBeanDefinitions(XmlBeanDefinitionReader.java:338)
2022-07-07T17:18:07.128185337Z 	at org.springframework.beans.factory.xml.XmlBeanDefinitionReader.loadBeanDefinitions(XmlBeanDefinitionReader.java:310)
2022-07-07T17:18:07.128189485Z 	at org.springframework.beans.factory.support.AbstractBeanDefinitionReader.loadBeanDefinitions(AbstractBeanDefinitionReader.java:196)
2022-07-07T17:18:07.128193313Z 	at org.springframework.beans.factory.support.AbstractBeanDefinitionReader.loadBeanDefinitions(AbstractBeanDefinitionReader.java:232)
2022-07-07T17:18:07.128197521Z 	at org.springframework.beans.factory.support.AbstractBeanDefinitionReader.loadBeanDefinitions(AbstractBeanDefinitionReader.java:203)
2022-07-07T17:18:07.128201759Z 	at org.springframework.web.context.support.XmlWebApplicationContext.loadBeanDefinitions(XmlWebApplicationContext.java:125)
2022-07-07T17:18:07.128205666Z 	at org.springframework.web.context.support.XmlWebApplicationContext.loadBeanDefinitions(XmlWebApplicationContext.java:94)
2022-07-07T17:18:07.128209153Z 	at org.springframework.context.support.AbstractRefreshableApplicationContext.refreshBeanFactory(AbstractRefreshableApplicationContext.java:130)
2022-07-07T17:18:07.128212710Z 	at org.springframework.context.support.AbstractApplicationContext.obtainFreshBeanFactory(AbstractApplicationContext.java:671)
2022-07-07T17:18:07.128215936Z 	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:553)
2022-07-07T17:18:07.128221787Z 	at org.springframework.web.servlet.FrameworkServlet.configureAndRefreshWebApplicationContext(FrameworkServlet.java:702)
2022-07-07T17:18:07.128226216Z 	at org.springframework.web.servlet.FrameworkServlet.createWebApplicationContext(FrameworkServlet.java:668)
2022-07-07T17:18:07.128233840Z 	at org.springframework.web.servlet.FrameworkServlet.createWebApplicationContext(FrameworkServlet.java:716)
2022-07-07T17:18:07.128237598Z 	at org.springframework.web.servlet.FrameworkServlet.initWebApplicationContext(FrameworkServlet.java:591)
2022-07-07T17:18:07.128241265Z 	at org.springframework.web.servlet.FrameworkServlet.initServletBean(FrameworkServlet.java:530)
2022-07-07T17:18:07.128245132Z 	at org.springframework.web.servlet.HttpServletBean.init(HttpServletBean.java:170)
2022-07-07T17:18:07.128249310Z 	at javax.servlet.GenericServlet.init(GenericServlet.java:180)
2022-07-07T17:18:07.128252847Z 	at org.eclipse.jetty.servlet.ServletHolder.initServlet(ServletHolder.java:632)
2022-07-07T17:18:07.128256584Z 	at org.eclipse.jetty.servlet.ServletHolder.initialize(ServletHolder.java:415)
2022-07-07T17:18:07.128260411Z 	at org.eclipse.jetty.servlet.ServletHandler.lambda$initialize$0(ServletHandler.java:731)
2022-07-07T17:18:07.128264279Z 	at java.base/java.util.stream.SortedOps$SizedRefSortingSink.end(SortedOps.java:357)
2022-07-07T17:18:07.128268086Z 	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:510)
2022-07-07T17:18:07.128271904Z 	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
2022-07-07T17:18:07.128275741Z 	at java.base/java.util.stream.StreamSpliterators$WrappingSpliterator.forEachRemaining(StreamSpliterators.java:310)
2022-07-07T17:18:07.128279618Z 	at java.base/java.util.stream.Streams$ConcatSpliterator.forEachRemaining(Streams.java:735)
2022-07-07T17:18:07.128283476Z 	at java.base/java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:762)
2022-07-07T17:18:07.128287393Z 	at org.eclipse.jetty.servlet.ServletHandler.initialize(ServletHandler.java:755)
2022-07-07T17:18:07.128291080Z 	at org.eclipse.jetty.servlet.ServletContextHandler.startContext(ServletContextHandler.java:379)
2022-07-07T17:18:07.128294848Z 	at org.eclipse.jetty.server.handler.ContextHandler.doStart(ContextHandler.java:910)
2022-07-07T17:18:07.128298585Z 	at org.eclipse.jetty.servlet.ServletContextHandler.doStart(ServletContextHandler.java:288)
2022-07-07T17:18:07.128302372Z 	at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:73)
2022-07-07T17:18:07.128305919Z 	at org.eclipse.jetty.util.component.ContainerLifeCycle.start(ContainerLifeCycle.java:169)
2022-07-07T17:18:07.128309536Z 	at org.eclipse.jetty.util.component.ContainerLifeCycle.doStart(ContainerLifeCycle.java:117)
2022-07-07T17:18:07.128313413Z 	at org.eclipse.jetty.server.handler.AbstractHandler.doStart(AbstractHandler.java:97)
2022-07-07T17:18:07.128317291Z 	at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:73)
2022-07-07T17:18:07.128321759Z 	at org.eclipse.jetty.util.component.ContainerLifeCycle.start(ContainerLifeCycle.java:169)
2022-07-07T17:18:07.128326017Z 	at org.eclipse.jetty.util.component.ContainerLifeCycle.doStart(ContainerLifeCycle.java:117)
2022-07-07T17:18:07.128334213Z 	at org.eclipse.jetty.server.handler.AbstractHandler.doStart(AbstractHandler.java:97)
2022-07-07T17:18:07.128338622Z 	at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:73)
2022-07-07T17:18:07.128342850Z 	at org.eclipse.jetty.util.component.ContainerLifeCycle.start(ContainerLifeCycle.java:169)
2022-07-07T17:18:07.128346517Z 	at org.eclipse.jetty.server.Server.start(Server.java:423)
2022-07-07T17:18:07.128350334Z 	at org.eclipse.jetty.util.component.ContainerLifeCycle.doStart(ContainerLifeCycle.java:110)
2022-07-07T17:18:07.128354202Z 	at org.eclipse.jetty.server.handler.AbstractHandler.doStart(AbstractHandler.java:97)
2022-07-07T17:18:07.128358029Z 	at org.eclipse.jetty.server.Server.doStart(Server.java:387)
2022-07-07T17:18:07.128361766Z 	at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:73)
2022-07-07T17:18:07.128365804Z 	at com.consol.citrus.http.server.HttpServer.startup(HttpServer.java:210)
2022-07-07T17:18:07.128369501Z 	at com.consol.citrus.server.AbstractServer.start(AbstractServer.java:93)
2022-07-07T17:18:07.128373448Z 	at com.consol.citrus.server.AbstractServer.initialize(AbstractServer.java:165)
2022-07-07T17:18:07.128377075Z 	at com.consol.citrus.http.server.HttpServer.initialize(HttpServer.java:219)
2022-07-07T17:18:07.128380893Z 	at com.consol.citrus.config.ComponentLifecycleProcessor.postProcessBeforeInitialization(ComponentLifecycleProcessor.java:59)
2022-07-07T17:18:07.128384780Z 	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.applyBeanPostProcessorsBeforeInitialization(AbstractAutowireCapableBeanFactory.java:440)
2022-07-07T17:18:07.128388898Z 	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1796)
2022-07-07T17:18:07.128392936Z 	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:620)
2022-07-07T17:18:07.128397144Z 	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:542)
2022-07-07T17:18:07.128401262Z 	at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:335)
2022-07-07T17:18:07.128405029Z 	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:234)
2022-07-07T17:18:07.128409818Z 	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:333)
2022-07-07T17:18:07.128413936Z 	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:208)
2022-07-07T17:18:07.128417854Z 	at org.springframework.beans.factory.support.DefaultListableBeanFactory.preInstantiateSingletons(DefaultListableBeanFactory.java:953)
2022-07-07T17:18:07.128421731Z 	at org.springframework.context.support.AbstractApplicationContext.finishBeanFactoryInitialization(AbstractApplicationContext.java:918)
2022-07-07T17:18:07.128429506Z 	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:583)
2022-07-07T17:18:07.128433674Z 	at org.springframework.test.context.support.AbstractGenericContextLoader.loadContext(AbstractGenericContextLoader.java:127)
2022-07-07T17:18:07.128437652Z 	at org.springframework.test.context.support.AbstractGenericContextLoader.loadContext(AbstractGenericContextLoader.java:60)
2022-07-07T17:18:07.128441339Z 	at org.springframework.test.context.support.AbstractDelegatingSmartContextLoader.delegateLoading(AbstractDelegatingSmartContextLoader.java:276)
2022-07-07T17:18:07.128445667Z 	at org.springframework.test.context.support.AbstractDelegatingSmartContextLoader.loadContext(AbstractDelegatingSmartContextLoader.java:244)
2022-07-07T17:18:07.128449635Z 	at org.springframework.test.context.cache.DefaultCacheAwareContextLoaderDelegate.loadContextInternal(DefaultCacheAwareContextLoaderDelegate.java:99)
2022-07-07T17:18:07.128453713Z 	at org.springframework.test.context.cache.DefaultCacheAwareContextLoaderDelegate.loadContext(DefaultCacheAwareContextLoaderDelegate.java:124)
2022-07-07T17:18:07.128457610Z 	at org.springframework.test.context.support.DefaultTestContext.getApplicationContext(DefaultTestContext.java:124)
2022-07-07T17:18:07.128461518Z 	at org.springframework.test.context.support.DependencyInjectionTestExecutionListener.injectDependencies(DependencyInjectionTestExecutionListener.java:118)
2022-07-07T17:18:07.128466036Z 	at org.springframework.test.context.support.DependencyInjectionTestExecutionListener.prepareTestInstance(DependencyInjectionTestExecutionListener.java:83)
2022-07-07T17:18:07.128470164Z 	at org.springframework.test.context.TestContextManager.prepareTestInstance(TestContextManager.java:248)
2022-07-07T17:18:07.128474272Z 	at org.springframework.test.context.testng.AbstractTestNGSpringContextTests.springTestContextPrepareTestInstance(AbstractTestNGSpringContextTests.java:149)
2022-07-07T17:18:07.128478270Z 	at com.consol.citrus.testng.spring.TestNGCitrusSpringSupport.beforeSuite(TestNGCitrusSpringSupport.java:207)
2022-07-07T17:18:07.128482338Z 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
2022-07-07T17:18:07.128486035Z 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
2022-07-07T17:18:07.128489892Z 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
2022-07-07T17:18:07.128493800Z 	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
2022-07-07T17:18:07.128497707Z 	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:132)
2022-07-07T17:18:07.128501444Z 	at org.testng.internal.MethodInvocationHelper.invokeMethodConsideringTimeout(MethodInvocationHelper.java:61)
2022-07-07T17:18:07.128505362Z 	at org.testng.internal.ConfigInvoker.invokeConfigurationMethod(ConfigInvoker.java:366)
2022-07-07T17:18:07.128508959Z 	at org.testng.internal.ConfigInvoker.invokeConfigurations(ConfigInvoker.java:320)
2022-07-07T17:18:07.128516263Z 	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:327)
2022-07-07T17:18:07.128520090Z 	at org.testng.SuiteRunner.run(SuiteRunner.java:286)
2022-07-07T17:18:07.128541080Z 	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:53)
2022-07-07T17:18:07.128550488Z 	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:96)
2022-07-07T17:18:07.128555007Z 	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1218)
2022-07-07T17:18:07.128558915Z 	at org.testng.TestNG.runSuitesLocally(TestNG.java:1140)
2022-07-07T17:18:07.128562842Z 	at org.testng.TestNG.runSuites(TestNG.java:1069)
2022-07-07T17:18:07.128566399Z 	at org.testng.TestNG.run(TestNG.java:1037)
2022-07-07T17:18:07.128570016Z 	at com.consol.citrus.testng.TestNGEngine.run(TestNGEngine.java:151)
2022-07-07T17:18:07.128573613Z 	at com.consol.citrus.main.CitrusApp.run(CitrusApp.java:125)
2022-07-07T17:18:07.128577510Z 	at com.consol.citrus.remote.controller.RunController.run(RunController.java:107)
2022-07-07T17:18:07.128581508Z 	at com.consol.citrus.remote.controller.RunController.runPackages(RunController.java:75)
2022-07-07T17:18:07.128585125Z 	at com.consol.citrus.remote.controller.RunController.runAll(RunController.java:60)
2022-07-07T17:18:07.128589153Z 	at com.consol.citrus.remote.CitrusRemoteApplication.runTests(CitrusRemoteApplication.java:262)
2022-07-07T17:18:07.128592780Z 	at com.consol.citrus.remote.CitrusRemoteApplication$1.run(CitrusRemoteApplication.java:215)
2022-07-07T17:18:07.128596657Z 	at com.consol.citrus.remote.job.RunJob.call(RunJob.java:43)
2022-07-07T17:18:07.128600645Z 	at com.consol.citrus.remote.job.RunJob.call(RunJob.java:29)
2022-07-07T17:18:07.128604953Z 	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
2022-07-07T17:18:07.128608931Z 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
2022-07-07T17:18:07.128612938Z 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
2022-07-07T17:18:07.128616645Z 	at java.base/java.lang.Thread.run(Thread.java:833)
2022-07-07T17:18:07.128620483Z Caused by: org.springframework.beans.factory.parsing.BeanDefinitionParsingException: Configuration problem: Unable to locate Spring NamespaceHandler for XML schema namespace [http://www.springframework.org/schema/util]
2022-07-07T17:18:07.128624611Z Offending resource: class path resource [com/consol/citrus/http/citrus-servlet-context.xml]
```

This xsd namespace is provided by `spring-beans` (which is implicititly on the classpath through `spring-context`, but adding it explicitly does not change the behaviour).

My assumption is that something goes south during the jar packaging.

--- a/citrus-remote-maven-plugin/src/main/resources/assemblies/test-jar.xml
+++ b/citrus-remote-maven-plugin/src/main/resources/assemblies/test-jar.xml
@@ -30,4 +30,9 @@
       <scope>test</scope>
     </dependencySet>
   </dependencySets>
+  <containerDescriptorHandlers>
+    <containerDescriptorHandler>
+      <handlerName>metaInf-spring</handlerName>
+    </containerDescriptorHandler>
+  </containerDescriptorHandlers>
 </assembly>

--- a/citrus-remote-maven-plugin/src/main/resources/assemblies/test-war.xml
+++ b/citrus-remote-maven-plugin/src/main/resources/assemblies/test-war.xml
@@ -44,4 +44,9 @@
       </includes>
     </fileSet>
   </fileSets>
+  <containerDescriptorHandlers>
+    <containerDescriptorHandler>
+      <handlerName>metaInf-spring</handlerName>
+    </containerDescriptorHandler>
+  </containerDescriptorHandlers>
 </assembly>

--- a/citrus-remote-sample/pom.xml
+++ b/citrus-remote-sample/pom.xml
@@ -27,6 +27,7 @@
         <citrus.remote.verify.phase>none</citrus.remote.verify.phase>
 
         <docker.build.phase>none</docker.build.phase>
+        <docker.copy.phase>none</docker.copy.phase>
         <docker.remove.phase>none</docker.remove.phase>
         <docker.start.phase>none</docker.start.phase>
         <docker.stop.phase>none</docker.stop.phase>
@@ -175,7 +176,7 @@
                         <goals>
                             <goal>copy</goal>
                         </goals>
-                        <phase>${citrus.remote.phase}</phase>
+                        <phase>${docker.copy.phase}</phase>
                     </execution>
                     <execution>
                         <id>stop-container</id>
@@ -311,6 +312,7 @@
                 <citrus.remote.verify.phase>verify</citrus.remote.verify.phase>
 
                 <docker.build.phase>pre-integration-test</docker.build.phase>
+                <docker.copy.phase>post-integration-test</docker.copy.phase>
                 <docker.remove.phase>post-integration-test</docker.remove.phase>
                 <docker.start.phase>pre-integration-test</docker.start.phase>
                 <docker.stop.phase>post-integration-test</docker.stop.phase>
@@ -323,6 +325,8 @@
             <properties>
                 <citrus.remote.test.phase>none</citrus.remote.test.phase>
                 <citrus.remote.verify.phase>none</citrus.remote.verify.phase>
+
+                <docker.copy.phase>none</docker.copy.phase>
             </properties>
         </profile>
         <profile>

--- a/citrus-remote-sample/pom.xml
+++ b/citrus-remote-sample/pom.xml
@@ -22,7 +22,9 @@
         <spring.version>6.1.12</spring.version>
 
         <!-- Project setup -->
-        <citrus.remote.phase>none</citrus.remote.phase>
+        <citrus.remote.report.directory>citrus-remote</citrus.remote.report.directory>
+        <citrus.remote.test.phase>none</citrus.remote.test.phase>
+        <citrus.remote.verify.phase>none</citrus.remote.verify.phase>
 
         <docker.build.phase>none</docker.build.phase>
         <docker.remove.phase>none</docker.remove.phase>
@@ -56,34 +58,6 @@
                         </goals>
                         <phase>package</phase>
                         <configuration>
-                            <assembly>
-                                <descriptor>
-                                    <inline>
-                                        <id>citrus-tests</id>
-                                        <formats>
-                                            <format>jar</format>
-                                        </formats>
-                                        <includeBaseDirectory>false</includeBaseDirectory>
-                                        <dependencySets>
-                                            <dependencySet>
-                                                <outputDirectory>/</outputDirectory>
-                                                <useProjectArtifact>false</useProjectArtifact>
-                                                <useProjectAttachments>true</useProjectAttachments>
-                                                <unpack>true</unpack>
-                                                <scope>test</scope>
-                                            </dependencySet>
-                                        </dependencySets>
-                                        <!-- Workaround for -->
-                                        <!-- https://github.com/citrusframework/citrus-remote/issues/2 -->
-                                        <!-- if the issue is fixed, the whole descriptor-section can be removed. -->
-                                        <containerDescriptorHandlers>
-                                            <containerDescriptorHandler>
-                                                <handlerName>metaInf-spring</handlerName>
-                                            </containerDescriptorHandler>
-                                        </containerDescriptorHandlers>
-                                    </inline>
-                                </descriptor>
-                            </assembly>
                             <mainClass>org.citrusframework.remote.sample.entrypoint.CustomEntrypoint</mainClass>
                         </configuration>
                     </execution>
@@ -91,9 +65,8 @@
                         <id>trigger-citrus-remote</id>
                         <goals>
                             <goal>test</goal>
-                            <goal>verify</goal>
                         </goals>
-                        <phase>${citrus.remote.phase}</phase>
+                        <phase>${citrus.remote.test.phase}</phase>
                         <configuration>
                             <server>
                                 <url>http://localhost:4567</url>
@@ -104,7 +77,19 @@
                                 <pollingInterval>15000</pollingInterval>
                             </run>
                             <report>
-                                <directory>citrus-remote</directory>
+                                <directory>${citrus.remote.report.directory}</directory>
+                            </report>
+                        </configuration>
+                    </execution>
+                    <execution>
+                        <id>verify-citrus-remote-tests</id>
+                        <goals>
+                            <goal>verify</goal>
+                        </goals>
+                        <phase>${citrus.remote.verify.phase}</phase>
+                        <configuration>
+                            <report>
+                                <directory>${citrus.remote.report.directory}</directory>
                             </report>
                         </configuration>
                     </execution>
@@ -322,7 +307,8 @@
         <profile>
             <id>docker</id>
             <properties>
-                <citrus.remote.phase>integration-test</citrus.remote.phase>
+                <citrus.remote.test.phase>integration-test</citrus.remote.test.phase>
+                <citrus.remote.verify.phase>verify</citrus.remote.verify.phase>
 
                 <docker.build.phase>pre-integration-test</docker.build.phase>
                 <docker.remove.phase>post-integration-test</docker.remove.phase>
@@ -335,7 +321,8 @@
         <profile>
             <id>skip-citrus-remote</id>
             <properties>
-                <citrus.remote.phase>none</citrus.remote.phase>
+                <citrus.remote.test.phase>none</citrus.remote.test.phase>
+                <citrus.remote.verify.phase>none</citrus.remote.verify.phase>
             </properties>
         </profile>
         <profile>

